app/src/main/java/io/appium/uiautomator2/server/ServerInstrumentation.java

```diff
@@ -50,13 +56,33 @@
     private long wakeLockTimeoutMs = 0;
     private boolean isServerStopped;
 
+    private static boolean shouldDisableSuppressAccessibilityService = Boolean.parseBoolean(
+            InstrumentationRegistry.getArguments().getString("DISABLE_SUPPRESS_ACCESSIBILITY_SERVICES", "false"));
+
+    private void setDefaultAutomationFlag() {
+        // Need to ensure enabling suppress accessibility service
+        Configurator.getInstance().setUiAutomationFlags(0);
+    }
+
+    private void disableSuppressAccessibilityService() {
+        // The flag is necessary not to stop running accessibility service
+        // https://developer.android.com/reference/android/app/UiAutomation
+        if (shouldDisableSuppressAccessibilityService && Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            Configurator.getInstance().setUiAutomationFlags(
+                    UiAutomation.FLAG_DONT_SUPPRESS_ACCESSIBILITY_SERVICES);
+        }
+    }
+
     private ServerInstrumentation(Context context, int serverPort) {
         if (!isValidPort(serverPort)) {
             throw new UiAutomator2Exception(String.format(
                     "The port is out of valid range [%s;%s]: %s", MIN_PORT, MAX_PORT, serverPort));
         }
         this.serverPort = serverPort;
         this.powerManager = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
+
+        setDefaultAutomationFlag();
```

Comment:
should we always reset these flags or only before we actually change them?

Reply:
Good point.
I continued to observe the lifecycle of `Configurator.getInstance()` after the latest commit.
The value could remain before/after uia2 server lifecycle.

So, changing the state explicitly is better.